marimo/_runtime/dataflow.py

```diff
@@ -196,11 +267,27 @@ def register_cell(self, cell_id: CellId_t, cell: CellImpl) -> None:
                     if name in self.definitions
                     else set()
                 ) - set((cell_id,))
+
+                # Handle SQL matching for hierarchical references
+                name_map: dict[Name, Name] = {}
+                sql_ref = cell.refs_data.get(name)
+                if (
```

Comment:
```python
other_ids_defining_name, variable_name = sql_ref.extract_hierarachial_matches(tables=self.table_defs, views=self.views_defs, catalog=self.catalog_defs) # Directly testable

...

@property
def variable_kinds(self) -> dict[Name, str]: # Name, kind
    ...

@property
def table_defs(self) -> set[Name]:
    return {name for name, kind in self.variable_kinds.items() if kind == "table"}

...
```

Like I said, just a bit of tidy- but I don't want this to be a major blocker. PR over all looks solid, I just think we can move some of the sql specific stuff directly to sqlref, and avoid the name splitting since sqlref contains that information.